File: gateway/src/main/java/com/example/gateway/config/HttpExchangeConfig.java
package com.example.gateway.config;

import com.example.gateway.exchange.DataeaseBackendExchange;
import lombok.RequiredArgsConstructor;
import org.springframework.cloud.client.loadbalancer.LoadBalanced;
import org.springframework.cloud.client.loadbalancer.reactive.ReactorLoadBalancerExchangeFilterFunction;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;
import org.springframework.web.reactive.function.client.support.WebClientAdapter;
import org.springframework.web.service.invoker.HttpServiceProxyFactory;

@Configuration
@RequiredArgsConstructor
public class HttpExchangeConfig {

    private final ReactorLoadBalancerExchangeFilterFunction reactorLoadBalancerExchangeFilterFunction;

    @Bean
    @LoadBalanced
    DataeaseBackendExchange dataeaseBackendExchange() {
        WebClient client = WebClient.builder()
                .filter(reactorLoadBalancerExchangeFilterFunction)
                .baseUrl("http://dataease-backend/de-api")
                .build();
        var factory = HttpServiceProxyFactory.builderFor(WebClientAdapter.create(client)).build();
        return factory.createClient(DataeaseBackendExchange.class);
    }

}

File: chat-analysis/src/main/kotlin/org/inet/aet/chatanalysis/config/HttpExchangeConfig.kt
package org.inet.aet.chatanalysis.config

import org.inet.aet.chatanalysis.exchange.api.DsWorkerExchange
import org.springframework.cloud.client.loadbalancer.LoadBalanced
import org.springframework.cloud.client.loadbalancer.reactive.ReactorLoadBalancerExchangeFilterFunction
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpStatusCode
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.support.WebClientAdapter
import org.springframework.web.service.invoker.HttpServiceProxyFactory

@Configuration
class HttpExchangeConfig (
    val reactorLoadBalancerExchangeFilterFunction: ReactorLoadBalancerExchangeFilterFunction
) {

    @Bean
    @LoadBalanced
    fun dsWorkerExchange(): DsWorkerExchange {
        val client = WebClient.builder()
            .filter(reactorLoadBalancerExchangeFilterFunction)
            .baseUrl("lb://ds-worker")
            .build()
        val factory = HttpServiceProxyFactory.builderFor(WebClientAdapter.create(client)).build()
        return factory.createClient(DsWorkerExchange::class.java)
    }

}